serenata_toolbox/chamber_of_deputies/presences_dataset.py

```diff
@@ -0,0 +1,173 @@
+import os
+import urllib
+import xml.etree.ElementTree as ET
+import socket
+import time
+
+import pandas as pd
+
+from serenata_toolbox.datasets.helpers import (
+    save_to_csv,
+    translate_column,
+    xml_extract_date,
+    xml_extract_datetime,
+    xml_extract_text,
+)
+
+class PresencesDataset:
+
+    URL = (
+        'http://www.camara.leg.br/SitCamaraWS/sessoesreunioes.asmx/ListarPresencasParlamentar'
+        '?dataIni={}'
+        '&dataFim={}'
+        '&numMatriculaParlamentar={}'
+    )
+
+    def fetch(self, deputies, start_date, end_date):
+        """
+        :param deputies: (pandas.DataFrame) a dataframe with deputies data
+        :param date_start: (str) date in the format dd/mm/yyyy
+        :param date_end: (str) date in the format dd/mm/yyyy
+        """
+        if os.environ.get('DEBUG') == '1':
+            print("Fetching data for {} deputies from {} -> {}".format(len(deputies), start_date, end_date))
+
+        records = self.__all_presences(deputies, start_date, end_date)
+
+        df = pd.DataFrame(records, columns=(
+            'term',
+            'congressperson_document',
+            'congressperson_name',
+            'party',
+            'state',
+            'date',
+            'present_on_day',
+            'justification',
+            'session',
+            'presence'
+        ))
+        return self.__translate(df)
+
+    def __all_presences(self, deputies, start_date, end_date):
+        error_count = 0
+        for i, deputy in deputies.iterrows():
+            if os.environ.get('DEBUG') == '1':
+                print(i, deputy.congressperson_name, deputy.congressperson_document)
+            url = self.URL.format(start_date, end_date, deputy.congressperson_document)
+            xml = self.__try_fetch_xml(10, url)
+
+            if xml == None:
+                error_count += 1
+            else:
+                root = ET.ElementTree(file=xml).getroot()
+                for presence in self.__parse_deputy_presences(root):
+                    yield presence
+
+            # Let's not slow down our integration tests
+            if os.environ.get('RUN_INTEGRATION_TESTS') != '1':
+                time.sleep(2)
```

Comment:
Yeah, theses `sleep`s have been around for a while 😊 The idea is to prevent us from hammering camara's API with requests. The reason behind having them baked into the code is that the first few times I ran these scripts their server would eventually start 💥 ing like half way through scraping.

I guess I could just pass it in as a parameter to the dataset constructor 🤔 Will give that a try